### ISSUE TITLE
GC thread exception when extracting corrupt wim file

### ISSUE THREAD

**Issue by svenclaesson** [microsoft/RecursiveExtractor#93]
Because of a bug in DiscUtils.Wim (see https://github.com/DiscUtils/DiscUtils/pull/263) extracting the attached image will throw an exception on the GC thread and bring the entire application down.

I have low hope in getting that PR merged since the maintainers of DiscUtils seams absent. 
Can you think of any other other workaround?

[TestDataCorruptWim.zip](https://github.com/microsoft/RecursiveExtractor/files/9831905/TestDataCorruptWim.zip)

**Comment by gfs** [comment]
You might have better luck getting your change integrated with this fork I saw discussed in the issues: https://github.com/LTRData/DiscUtils.

That said, LTRData isn't producing builds to nuget such that I could consume them. I have no objection to switching to a package with more frequent maintenance - or a different package entirely, but it would have to be off Nuget. 

I cannot independently vouch for this fork, I have just noticed it exists and appears to be more active.

I took a quick look at your sample file and I don't think there's any changes I can make in Recursive Extractor to account for it, but if you can think of anything I'm willing to consider adding a workaround.

**Comment by svenclaesson** [comment]
I see now that LTRData publish nuget packages. 
Tried it and it fixes this [bug](https://github.com/LTRData/DiscUtils/blob/LTRData.DiscUtils-initial/Library/DiscUtils.Wim/WimFileSystem.cs#L620). Also fixes another VHDX hang bug i recently came across so it would be very useful to migrate to use this fork instead. There are also many good memory allocation optimizations.
One issue is that it seams to break a few WIM tests, because of other reasons then dispose, but i have not had the chance to dig deeper yet.

**Comment by gfs** [comment]
@svenclaesson 

Thanks for the heads up on the availability of the nuget. I may have time to experiment with migrating in the next few weeks, sounds like there are some good improvements to be had.

**Comment by gfs** [comment]
I've confirmed your findings that it seems to mostly be a drop in replacement via running the tests except for the WIM format (I did notice a smaller number of test failures with zips as well, but haven't fully investigated yet). I think its likely something resolvable but I'll have to revisit in the next few weeks to figure out how to work around the issue here. Opened issue #144 to track work on migrating to the LTRData package.
